components/supervisor/pkg/ports/ports.go

```diff
@@ -360,8 +360,6 @@ func (pm *Manager) nextState(ctx context.Context) map[uint32]*managedPort {
 				state[port] = mp
 			}
 			mp.LocalhostPort = port
```

Comment:
I think it is important to keep it here, step 2 should be about handling coinfgs

Reply:
You can add some debug logs and enable them `#SUPERVISOR_DEBUG_ENABLE=true/...` prefix, then you can check workspace logs.